### PR TITLE
test: improve TypeScript test coverage

### PR DESCRIPTION
**Pull request by TomChv**
- Move Typescript related tests into its own files.
- Add `TestModuleTypescriptSignaturesBuildinTypes`
- Add `TestModuleTypescriptSignatureUnexported`
- Add `TestModuleTypescriptDocs`
- Add `TestModuleTypescriptOptional`
- Add `TestModuleTypescriptWithOtherModuleTypes`

*Additional notes*
- Fix an issue with variadic arguments not correctly forwarded

💡 This is the final PR focused on increasing Typescript SDK coverage.

Fixes DEV-3310
Fixes DEV-3147

### CODE REVIEW

**Review by gerhard — APPROVED** [review]
This looks good to me. I know that it needs a rebase before it can be merged, but if you want to do that @TomChv, I think that this is OK to flow through into `main`, and then follow-up with anything that @helderco may flag.

**Comment by helderco** [comment]
I'm prioritizing some urgent last-minute changes in the CLI before release. It takes time to review this properly, so I plan to do it right after. There is no rush to merge this before the next release, as the only user-facing change is the added syntax sugar (variadic arguments).

**Comment by gerhard** [comment]
OK @helderco, leaving this with you & removing it from the next release milestone.

File: sdk/typescript/introspector/test/scan.spec.ts
Comment (TomChv) [reply sdk/typescript/introspector/test/scan.spec.ts:40]: > 1. Test individual parts in separate test cases
Instead of asserting the whole structure, assert individual fields that you care about in tests. This is what we do mostly in integration tests, for example, with module introspection and JSON querying. Then if you add a new field, you don't have to update all the test cases, just the new test that queries for it specifically.

I agree with you but the thing is that it will take as much time to specify specific field inside the `assert`.
And testing everything was actually useful because I found issues in other cases in #6537

I'm down to work on the golden part in a follow up though, I'll need more details to understand this

File: core/integration/module_typescript_test.go
Comment (TomChv) [reply core/integration/module_typescript_test.go:387]: Fixed in [`81bca97` (#6485)](https://github.com/dagger/dagger/pull/6485/commits/81bca979e1f146513cd0a960894d58a29b2f9c0f)

File: sdk/typescript/introspector/test/invoke.spec.ts
Comment (TomChv) [reply sdk/typescript/introspector/test/invoke.spec.ts]: Good eyes! Fixed in [`81bca97` (#6485)](https://github.com/dagger/dagger/pull/6485/commits/81bca979e1f146513cd0a960894d58a29b2f9c0f)

File: core/integration/module_typescript_test.go
Comment (TomChv) [reply core/integration/module_typescript_test.go]: Fixed in [`81bca97` (#6485)](https://github.com/dagger/dagger/pull/6485/commits/81bca979e1f146513cd0a960894d58a29b2f9c0f), I removed `t.Parallel()` where it wasn't necessary

File: core/integration/module_typescript_test.go
Comment (TomChv) [reply core/integration/module_typescript_test.go]: Fixed in [`81bca97` (#6485)](https://github.com/dagger/dagger/pull/6485/commits/81bca979e1f146513cd0a960894d58a29b2f9c0f)

File: core/integration/module_typescript_test.go
Comment (TomChv) [reply core/integration/module_typescript_test.go:14]: Fixed in [`81bca97` (#6485)](https://github.com/dagger/dagger/pull/6485/commits/81bca979e1f146513cd0a960894d58a29b2f9c0f)

File: sdk/typescript/introspector/test/invoke.spec.ts
Comment (helderco) [inline sdk/typescript/introspector/test/invoke.spec.ts:236]: This setup doesn't need to be repeated. You could use a [`before` hook](https://mochajs.org/#hooks):

```ts
let scanResult: ScanResult

before(function () {
    ...
    scanResult = scan(files)
})

it(..., function () {
    invoke(scanResult)
})
```

**Review by helderco — APPROVED** [review]

File: sdk/typescript/introspector/test/invoke.spec.ts
Comment (TomChv) [reply sdk/typescript/introspector/test/invoke.spec.ts:236]: I'll update the in a follow up PR

**Pull request merged**
